tools/java/org/unicode/cldr/util/XMLSource.java

```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
+                    if (currentFullXPath.startsWith("//ldml/layout/orientation")) {
+                        XPathParts parts = XPathParts.getFrozenInstance(currentFullXPath);
+                        String value = parts.getAttributeValue(-1, "characters");
+                        if (value != null) {
+                            addPath("//ldml/layout/orientation/characterOrder", value);
+                            skipAdd = true;
+                        }
+                        value = parts.getAttributeValue(-1, "lines");
+                        if (value != null) {
+                            addPath("//ldml/layout/orientation/lineOrder", value);
+                            skipAdd = true;
+                        }
+                    }
+                    if (!skipAdd) {
+                        addPath(currentFullXPath, lastChars);
+                    }
+                    lastLeafNode = lastActiveLeafNode = currentFullXPath;
+                }
+                lastChars = "";
+            } else {
+                Log.logln(LOG_PROGRESS && lastActiveLeafNode != null, "pop: zeroing last leafNode: "
+                    + lastActiveLeafNode);
+                lastActiveLeafNode = null;
+                if (comment != null) {
+                    source.addComment(lastLeafNode, comment, XPathParts.Comments.CommentType.POSTBLOCK);
+                    comment = null;
+                }
+            }
+            // currentXPath = stripAfter(currentXPath, qName);
+            currentFullXPath = stripAfter(currentFullXPath, qName);
+            justPopped = true;
+        }
+
+        static Pattern WHITESPACE_WITH_LF = PatternCache.get("\\s*\\u000a\\s*");
+        Matcher whitespaceWithLf = WHITESPACE_WITH_LF.matcher("");
+        static final UnicodeSet CONTROLS = new UnicodeSet("[:cc:]");
+
+        /**
+         * Trim leading whitespace if there is a linefeed among them, then the same with trailing.
+         *
+         * @param source
+         * @return
+         */
+        private String trimWhitespaceSpecial(String source) {
+            if (DEBUG && CONTROLS.containsSome(source)) {
+                System.out.println("*** " + source);
+            }
+            if (!source.contains("\n")) {
+                return source;
+            }
+            source = whitespaceWithLf.reset(source).replaceAll("\n");
+            return source;
+        }
+
+        private void warnOnOverride(String former, String formerPath) {
+            String distinguishing = CLDRFile.getDistinguishingXPath(formerPath, null);
+            System.out.println("\tERROR in " + source.getLocaleID()
+                + ";\toverriding old value <" + former + "> at path " + distinguishing +
+                "\twith\t<" + lastChars + ">" +
+                CldrUtility.LINE_SEPARATOR + "\told fullpath: " + formerPath +
+                CldrUtility.LINE_SEPARATOR + "\tnew fullpath: " + currentFullXPath);
+            overrideCount += 1;
```

Comment:
`overrideCount++`

Reply:
compiler will do the same with these, so just a matter of style.